src/contrast.rs

```diff
@@ -266,6 +268,90 @@ pub fn equalize_histogram(image: &GrayImage) -> GrayImage {
     out
 }
 
+/// Normalizes contrast of a Grayscale image image in place such that pixel brightness (Min,Max) maps to (new_Min,new_Max)
+/// See also  [Normalizes Equalization (wikipedia)] https://en.wikipedia.org/wiki/Normalization_(image_processing)
+pub fn normalize_linear(image: &GrayImage, new_min: u8, new_max: u8) -> GrayImage {
+    let mut out = image.clone();
+    normalize_linear_mut(&mut out, new_min, new_max);
+    out
+}
+
+/// Normalizes contrast of a Grayscale image such that pixel brightness (Min,Max) maps to (new_Min,new_Max)
+/// See also  [Normalizes Equalization (wikipedia)] https://en.wikipedia.org/wiki/Normalization_(image_processing)
+pub fn normalize_linear_mut(image: &mut GrayImage, new_min: u8, new_max: u8) {
+    assert!(
+        new_max > new_min,
+        "new_max must be strictly greater than new_min"
+    );
+
+    if image.is_empty() {
+        return;
+    }
+
+    let histogram = histogram(image).channels[0];
+    let min = histogram
+        .iter()
+        .enumerate()
+        .find_map(|(i, x)| if *x != 0 { Some(i as u8) } else { None })
+        .unwrap();
+    let max = histogram
+        .iter()
+        .enumerate()
+        .rev()
+        .find_map(|(i, x)| if *x != 0 { Some(i as u8) } else { None })
+        .unwrap();
+
+    //equation from https://en.wikipedia.org/wiki/Normalization_(image_processing)
+    let fraction = (new_max - new_min) as f64 / (max - min) as f64;
+    let f = |_, _, pixel: Luma<u8>| {
+        Luma([(((pixel.0[0] - min) as f64 * fraction) + new_min as f64) as u8])
+    };
+
+    map_pixels_mut(image, f);
```

Comment:
These closures are not always optimised away by the compiler. I’d be interested in seeing a benchmark for a raw loop vs this closure based version.

Reply:
I benchmarked using `map_pixels_mut()` with the closure vs a direct `for` loop:
Closure:
```
test contrast::benches::bench_scale_linear     ... bench:      51,119 ns/iter (+/- 308)
test contrast::benches::bench_scale_linear_mut ... bench:      50,364 ns/iter (+/- 192)
```
For Loop:
```
test contrast::benches::bench_scale_linear     ... bench:      51,119 ns/iter (+/- 308)
test contrast::benches::bench_scale_linear_mut ... bench:      50,364 ns/iter (+/- 192)
```

Seems identical to me, but maybe I'm benchmarking it wrong.